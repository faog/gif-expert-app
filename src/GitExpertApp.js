import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Manzana', 'Naranja', 'Pera']);

  return(
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />
      <ol>
        {
          categories.map( category => {
            return <li key= { category }> { category }</li>
          })
        }
      </ol>
    </>
  )
}

export default GifExpertApp;
