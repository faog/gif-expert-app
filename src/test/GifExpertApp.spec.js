import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GitExpertApp';

describe('<GifExpertApp /> Component', () => {
  test('should be show component', () => {
    const wrapper = shallow(<GifExpertApp  />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be show categories list', () => {
    const categories = ['Manzana', 'Naranja', 'Pera'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
