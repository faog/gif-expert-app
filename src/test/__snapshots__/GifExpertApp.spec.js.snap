// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> Component should be show categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Manzana"
    key="Manzana"
  />
  <GifGrid
    category="Naranja"
    key="Naranja"
  />
  <GifGrid
    category="Pera"
    key="Pera"
  />
</Fragment>
`;

exports[`<GifExpertApp /> Component should be show component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
